### DISCUSSION TITLE
Customize library columns in pnp search webpart

### DISCUSSION THREAD

**Discussion by Shinsha554** [microsoft-search/pnp-modern-search#3858] [Q&A] [answered]
I have created a document library and have defined some column. I am able to get the documents in pnp search webpart, but not able to see the columns I have created.
My requirement is to have a pnp filter webpart (filters will be my column name). 
I have created the custome property in selected property of search result webpart. But how to map that with document library column

**Comment by kasperbolarsen** [comment, marked as answer]
Rule no 1, the columns should be created as site columns,  no list columns. 
Rule no 2, you must add a document or list item that contains a value in each column
Rule no 3, you must map the crawled properties to managed properties,  and it must be mapped to a RefinableString/Int/Date in order to use it as a refiner.
Rule no 4, the managed properties must be added to "Selected Properties " in the Results web part before you can use it in the refiner.

**Reply by Shinsha554** [reply]
@kasperbolarsen Thank you so much, it got worked, I am a beginner to this PnP .
I have another query , in search result webpart I displayed my contents in tiles (selected card layout) so is it possible to add a download icon there? So that in one click we can download the docs

**Reply by kasperbolarsen** [reply]
Please check the documentation,  it should be possible as far as I know,  and documented as well

**Reply by Shinsha554** [reply]
@kasperbolarsen 
Where can I find the documentation?
Also when I try to display my custom fields in card layout , I am not able to see the add button in edit result field. But for list layout I can easily add my custom fields

**Reply by patrikhellgren** [reply]
@Shinsha554 I have written a blog post on how to use the download feature in the results webpart, [https://www.patrikhellgren.com/blog/how-to-use-the-pnp-modern-search-download-feature/](https://www.patrikhellgren.com/blog/how-to-use-the-pnp-modern-search-download-feature/)

**Reply by Shinsha554** [reply]
@patrikhellgren I hope this is for list layout, I have chosen card layout. I would like to build something like below, is it possible using pnp modern webparts?
![20240712_131625](https://github.com/user-attachments/assets/34a183ac-e0ac-4573-8651-d5748c4bc3f3)

**Reply by kasperbolarsen** [reply]
You will have to enable that cards can be selected before download becomes an option

**Reply by Shinsha554** [reply]
@kasperbolarsen , @patrikhellgren  I have created one sample result item with download icon, on click of this icon, I need to download the item specified in that tile. Is there is any way to get that link.

What I did is : I have created one site column for download and in my document library created direct link for downloading the item. And used that column as href for download icon. But when I click on that download option it is opening the document, instead of direct download. 
![17211435474667992985569186273218](https://github.com/user-attachments/assets/bf6133e2-855f-4634-ba74-431e0e9e693a)

**Reply by Shinsha554** [reply]
> @kasperbolarsen , @patrikhellgren I have created one sample result item with download icon, on click of this icon, I need to download the item specified in that tile. Is there is any way to get that link.
> 
> What I did is : I have created one site column for download and in my document library created direct link for downloading the item. And used that column as href for download icon. But when I click on that download option it is opening the document, instead of direct download. ![17211435474667992985569186273218](https://private-user-images.githubusercontent.com/175245204/349177083-bf6133e2-855f-4634-ba74-431e0e9e693a.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjExNDM5MTgsIm5iZiI6MTcyMTE0MzYxOCwicGF0aCI6Ii8xNzUyNDUyMDQvMzQ5MTc3MDgzLWJmNjEzM2UyLTg1NWYtNDYzNC1iYTc0LTQzMWUwZTllNjkzYS5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxNlQxNTI2NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0wN2QyZjQ5ZTFmZThjNmVlN2IxOTBlZDgwNmZkN2IwYTY1ZjhjNWNlZjhiZmI3NTcwNDU5MmUxNTY5OWM5MjhlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.i-NziX3hnYx5h5F2tKDBXC1BiqLQtZ28iNqJIQ486bg)


Also  how to rearrange the fields in tile, currently my tile is look like below, but I need that download icon in bottom right corner, and need some spaces between the tags, also don't need the file icon portion, I removed the file icon and the preview but that case  the space remains the same ,only pictures getting removed
![1721143805719723061337383131943](https://github.com/user-attachments/assets/8ac82d99-6b36-44be-ba34-fc87801203de)

**Comment by kasperbolarsen** [comment]
https://microsoft-search.github.io/pnp-modern-search/

tors. 11. jul. 2024 19.18 skrev Shinsha554 ***@***.***>:

**Comment by Shinsha554** [comment]
Hi ,
Is there is any way to rearrange the components inside the tile? Also is it possible to remove the file icon and image portion? I am able to remove the file icon and images but the section is still there. I need to remove the upper portion of the card layout.
Could you please help here.

**Reply by kasperbolarsen** [reply]
Please provide more info, preferably some screenshots as it is unclear what you are referring to

**Reply by Shinsha554** [reply]
Below is my tile, I need to add download option in the bottom right corner, and need to remove upper part of the tile (here I can see that tile is divided into two, I just need the lower part)
![17229329475606303627705974848404](https://github.com/user-attachments/assets/e43758ae-22c7-482d-9d6a-48fabd89651b)

**Reply by kasperbolarsen** [reply]
Please ask this question in a new post as this post has been answered

**Reply by Shinsha554** [reply]
Actually I didn't get the answer, that is why again posted.
